### ISSUE TITLE
Removed node added again into the graph

### ISSUE THREAD

**Issue by sachinsngh165** [P2PSP/simulator#36]

**Comment by sachinsngh165** [comment]
A quick solution would be to put NODE OUT line in drawing file whenever a peer receives good bye message from the splitter. Also, respond to request message if and only if the sender is in peer list.

**Comment by sachinsngh165** [comment]
@cristobalmedinalopez, is there any scenario when above solution would fail?

**Comment by sachinsngh165** [comment]
What should we do for following scenario:
When peer receives a `request` messaege from removed peer before receiving `good bye` message  from the splitter (edited). 
if we follow this solution:
>@cristobalmedinalopez A quick solution would be to put NODE OUT line in drawing file whenever a peer receives good bye message from the splitter. Also, respond to request message if and only if the sender is in peer list.
then removed peer would appear for time until it don't receives a good bye message.

**Comment by cristobalmedinalopez** [comment]
>A quick solution would be to put NODE OUT line in drawing file whenever a peer receives good bye message from the splitter.
You mean it removes itself, isn't it?

>Also, respond to request message if and only if the sender is in peer list.
Yes, it sounds good!

>When peer receives a request messaege from removed peer before receiving good bye message from the splitter (edited).
Does that action trigger the writing of a NODE IN line in the drawing file?

**Comment by sachinsngh165** [comment]
>Does that action trigger the writing of a NODE IN line in the drawing file?

yes, whenever a peer receives a request message from another, and if it is not in the forwarding list of origin (of requested chunk), then it write NODE IN  for sender in drawing file, that's the cause of this bug. This would overwrite splitter's action of NODE OUT.

**Comment by sachinsngh165** [comment]
> >A quick solution would be to put NODE OUT line in drawing file whenever a peer receives good bye message from the splitter.

>You mean it removes itself, isn't it?

No, i was talking in reference to other peers who will receive GOODBYE message from removed peers.

**Comment by cristobalmedinalopez** [comment]
>No, i was talking in reference to other peers who will receive GOODBYE message from removed peers.

It could work with honest peers.

**Comment by sachinsngh165** [comment]
Even checking sender in peer_list before responding to request_chunk message will not work due to above reason.

**Comment by cristobalmedinalopez** [comment]
I think it's not an important issue that removed peers appear like a ghost if finally they are completely removed from the graph. What do you think @vicente-gonzalez-ruiz? Any other idea @josmanual?

**Comment by sachinsngh165** [comment]
@cristobalmedinalopez , how should we check if removed peer is in peer_list or not, as we just have a forwarding list? should we check this peer in every forwarding list ?
